### PR TITLE
Add LIMIT 1 to coverimage subquery to get correct image using MariaDB

### PR DESCRIPTION
**Pull request by kurtaschliman**
Add LIMIT 1 to subquery to make ORDER BY work with MariaDB. MariaDB ignores ORDER BY in a subquery unless LIMIT is used. Reference: https://mariadb.com/kb/en/why-is-order-by-in-a-from-subquery-ignored/.  This provides a workaround using LIMIT, but could possibly use a further refactor to not use the subquery at all.

### CODE REVIEW

**Comment by galinski** [comment]
@kurtaschliman have you find any "good" solution for this problem?

**Comment by kurtaschliman** [comment]
@galinski I don't believe I ever found a "good" solution other than that proposed. I haven't done as much work with Shopp lately so I haven't been dealing with this issue recently.

File: core/model/Product.php [hunk 1]
```diff
@@ -283,7 +283,7 @@ public function load_coverimages ( $ids ) {
 		if ( empty($ids) ) return;
 		$table = ShoppDatabaseObject::tablename(ShoppMetaObject::$table);
 		$sortorder = $this->image_order();
-		sDB::query("SELECT * FROM ( SELECT * FROM $table WHERE context='product' AND type='image' AND parent IN ($ids) ORDER BY $sortorder ) AS img GROUP BY parent",'array',array($this, 'metasetloader'),'parent','metatype','name',false);
+		sDB::query("SELECT * FROM ( SELECT * FROM $table WHERE context='product' AND type='image' AND parent IN ($ids) ORDER BY $sortorder LIMIT 18446744073709551615 ) AS img GROUP BY parent",'array',array($this, 'metasetloader'),'parent','metatype','name',false);
```
Comment (msigley) [inline core/model/Product.php]: Why don't we move the order by to the parent query like the MariaDB article suggests:
```
DB::query("SELECT * FROM ( SELECT * FROM $table WHERE context='product' AND type='image' AND parent IN ($ids) ) AS img ORDER BY img.$sortorder GROUP BY parent",'array',array($this,'metaloader'),'parent','metatype','name',false);
```
That seems to work under my MariaDB 10 environment.

**Comment by jonathandavis** [comment]
Won't this kill MySQL in 32-bit environments?

**Comment by jonathandavis** [comment]
Unfortuantely @msigley that does not work in MySQL.

**Comment by msigley** [comment]
I would like to refactor this to remove the use of the sub query for images with "ORDER BY sortorder ASC" because it could be easily replaced with a "WHERE ... AND sortorder='0'".

I'll submit another PR tomorrow for this optimization.

**Comment by jonathandavis** [comment]
@msigley It is already refactored in 2f7615770f1b9eda046c0f68d5d37c63b126d13a

**Pull request merged**